src/integrity-checker.js

```diff
@@ -171,16 +238,27 @@ export default class InstallationIntegrityChecker {
     }
 
     const linkedModules = this.config.linkedModules;
+
     if (linkedModules.length) {
       result.linkedModules = linkedModules.sort(sortAlpha);
     }
 
-    Object.keys(lockfile).forEach(key => {
+    for (const key of Object.keys(lockfile)) {
       result.lockfileEntries[key] = lockfile[key].resolved || '';
-    });
+    }
+
+    for (const modulesFolder of this._getModulesFolders({workspaceLayout})) {
+      if (await fs.exists(modulesFolder)) {
```

Comment:
The `_getModulesFolders` function only returns the theoretical folders, it does not check that those folders actually exist on the disk. So for example, if we have a workspace named `foobar`, the function will return `foobar/node_modules` without checking that this folder actually exists.